From 4c79d814e6f4e71ccc4ebd00b1503d7a41b2f531 Mon Sep 17 00:00:00 2001
From: Billy Tsai <billy_tsai@aspeedtech.com>
Date: Wed, 2 Mar 2022 11:15:04 +0800
Subject: [PATCH] sig_def_list/ast10x0 : Fix the register setting of gpio.

Follow the spec. ast1030_v0.7.pdf chapter 5.1 to fix the register
setting of the GPIO pinctrl.

Signed-off-by: Billy Tsai <billy_tsai@aspeedtech.com>
Change-Id: I1b8c35f92e34fe6927aada195c0f49121184cead
---
 soc/arm/aspeed/ast10x0/sig_def_list.h | 46 ++++++++++-----------------
 1 file changed, 16 insertions(+), 30 deletions(-)

diff --git a/soc/arm/aspeed/ast10x0/sig_def_list.h b/soc/arm/aspeed/ast10x0/sig_def_list.h
index f5ac13abf8..d729c2e48c 100644
--- a/soc/arm/aspeed/ast10x0/sig_def_list.h
+++ b/soc/arm/aspeed/ast10x0/sig_def_list.h
@@ -222,40 +222,26 @@ SIG_DEFINE(SDA14, G16, SIG_DESC_CLEAR(0x418, 23), SIG_DESC_SET(0x4B8, 23))
 
 #ifdef CONFIG_GPIO_ASPEED
 /* UART1 */
-GPIO_SIG_DEFINE(GPIOL0, F12, SIG_DESC_CLEAR(0x418, 24))
-GPIO_SIG_DEFINE(GPIOL1, E12, SIG_DESC_CLEAR(0x418, 25))
+GPIO_SIG_DEFINE(GPIOL0, F12, SIG_DESC_CLEAR(0x418, 24), SIG_DESC_CLEAR(0x51C, 8))
+GPIO_SIG_DEFINE(GPIOL1, E12, SIG_DESC_CLEAR(0x418, 25), SIG_DESC_CLEAR(0x51C, 8))
 /* UART6 */
-GPIO_SIG_DEFINE(GPIOM0, B15, SIG_DESC_CLEAR(0x41C, 0))
-GPIO_SIG_DEFINE(GPIOM1, E11, SIG_DESC_CLEAR(0x41C, 1))
+GPIO_SIG_DEFINE(GPIOM0, B15, SIG_DESC_CLEAR(0x41C, 0), SIG_DESC_CLEAR(0x4BC, 0))
+GPIO_SIG_DEFINE(GPIOM1, E11, SIG_DESC_CLEAR(0x41C, 1), SIG_DESC_CLEAR(0x4BC, 1))
 /* UART7 */
-GPIO_SIG_DEFINE(GPIOM2, D14, SIG_DESC_CLEAR(0x41C, 2))
-GPIO_SIG_DEFINE(GPIOM3, C14, SIG_DESC_CLEAR(0x41C, 3))
+GPIO_SIG_DEFINE(GPIOM2, D14, SIG_DESC_CLEAR(0x41C, 2), SIG_DESC_CLEAR(0x4BC, 2))
+GPIO_SIG_DEFINE(GPIOM3, C14, SIG_DESC_CLEAR(0x41C, 3), SIG_DESC_CLEAR(0x4BC, 3))
 /* UART8 */
-GPIO_SIG_DEFINE(GPIOM4, A16, SIG_DESC_CLEAR(0x41C, 4))
-GPIO_SIG_DEFINE(GPIOM5, A15, SIG_DESC_CLEAR(0x41C, 5))
+GPIO_SIG_DEFINE(GPIOM4, A16, SIG_DESC_CLEAR(0x41C, 4), SIG_DESC_CLEAR(0x4BC, 4))
+GPIO_SIG_DEFINE(GPIOM5, A15, SIG_DESC_CLEAR(0x41C, 5), SIG_DESC_CLEAR(0x4BC, 5))
 /* ESPI */
-GPIO_SIG_DEFINE(GPION4, G14, SIG_DESC_CLEAR(0x41C, 12), SIG_DESC_CLEAR(0x4BC, 20))
-GPIO_SIG_DEFINE(GPION5, F16, SIG_DESC_CLEAR(0x41C, 13), SIG_DESC_CLEAR(0x4BC, 21))
-GPIO_SIG_DEFINE(GPION6, F15, SIG_DESC_CLEAR(0x41C, 14), SIG_DESC_CLEAR(0x4BC, 22))
-GPIO_SIG_DEFINE(GPION7, F14, SIG_DESC_CLEAR(0x41C, 15), SIG_DESC_CLEAR(0x4BC, 23))
-GPIO_SIG_DEFINE(GPIOO0, E16, SIG_DESC_CLEAR(0x41C, 16), SIG_DESC_CLEAR(0x4BC, 24))
-GPIO_SIG_DEFINE(GPIOO1, E15, SIG_DESC_CLEAR(0x41C, 17), SIG_DESC_CLEAR(0x4BC, 25))
-GPIO_SIG_DEFINE(GPIOO2, E14, SIG_DESC_CLEAR(0x41C, 18), SIG_DESC_CLEAR(0x4BC, 26))
-GPIO_SIG_DEFINE(GPIOO3, E13, SIG_DESC_CLEAR(0x41C, 19), SIG_DESC_CLEAR(0x4BC, 27))
-/* FWSPI */
-GPIO_SIG_DEFINE(GPIOR2, D8, SIG_DESC_CLEAR(0x430, 10))
-GPIO_SIG_DEFINE(GPIOR3, D5, SIG_DESC_CLEAR(0x430, 11))
-/* SPI1 */
-GPIO_SIG_DEFINE(GPIOS1, A4, SIG_DESC_CLEAR(0x430, 17))
-GPIO_SIG_DEFINE(GPIOS2, B4, SIG_DESC_CLEAR(0x430, 18))
-/* SPI2 */
-GPIO_SIG_DEFINE(GPIOP6, A10, SIG_DESC_CLEAR(0x41C, 30))
-GPIO_SIG_DEFINE(GPIOP7, B10, SIG_DESC_CLEAR(0x41C, 31))
-GPIO_SIG_DEFINE(GPIOQ0, A9, SIG_DESC_CLEAR(0x430, 0))
-GPIO_SIG_DEFINE(GPIOQ1, B9, SIG_DESC_CLEAR(0x430, 1))
-GPIO_SIG_DEFINE(GPIOQ2, C9, SIG_DESC_CLEAR(0x430, 2))
-GPIO_SIG_DEFINE(GPIOQ3, D9, SIG_DESC_CLEAR(0x430, 3))
-GPIO_SIG_DEFINE(GPIOQ4, A8, SIG_DESC_CLEAR(0x430, 4))
+GPIO_SIG_DEFINE(GPION4, G14, SIG_DESC_CLEAR(0x41C, 12))
+GPIO_SIG_DEFINE(GPION5, F16, SIG_DESC_CLEAR(0x41C, 13))
+GPIO_SIG_DEFINE(GPION6, F15, SIG_DESC_CLEAR(0x41C, 14))
+GPIO_SIG_DEFINE(GPION7, F14, SIG_DESC_CLEAR(0x41C, 15))
+GPIO_SIG_DEFINE(GPIOO0, E16, SIG_DESC_CLEAR(0x41C, 16))
+GPIO_SIG_DEFINE(GPIOO1, E15, SIG_DESC_CLEAR(0x41C, 17))
+GPIO_SIG_DEFINE(GPIOO2, E14, SIG_DESC_CLEAR(0x41C, 18))
+GPIO_SIG_DEFINE(GPIOO3, E13, SIG_DESC_CLEAR(0x41C, 19))
 /* ADC */
 GPIO_SIG_DEFINE(GPIT0, B2, SIG_DESC_SET(0x430, 24))
 GPIO_SIG_DEFINE(GPIT1, B1, SIG_DESC_SET(0x430, 25))
-- 
2.25.1

